toolkit/tools/pkg/simplechroottool/simplechroottool.go

```diff
@@ -0,0 +1,146 @@
+//   - chrootSpecDir: The path to the SPECS directory inside the chroot "/SPECS."
+
+type SimpleChrootTool struct {
+	BuildDirPath  string
+	WorkerTarPath string
+	chroot        *safechroot.Chroot
+	chrootSpecDir string
+}
+
+// Chroot returns the chroot environment we are working in, call Chroot().Run() to execute commands inside the chroot. Call
+// InitializeChroot() before calling this function.
+func (s *SimpleChrootTool) Chroot() *safechroot.Chroot {
+	return s.chroot
+}
+
+// ChrootSpecDir returns the directory inside the choot where the specs dir is mounted. Call InitializeChroot() before calling this function.
+func (s *SimpleChrootTool) ChrootSpecDir() string {
+	return s.chrootSpecDir
+}
+
+// InitializeChroot initializes the chroot environment so .Run() can be used to execute commands inside the chroot. This function
+// should be called before any other functions in this package.
+//   - chrootName: The name of the chroot directory to use
+//   - specsDirPath: The path to the directory containing the spec files
+func (s *SimpleChrootTool) InitializeChroot(chrootName, specsDirPath string) (err error) {
+	const (
+		existingDir = false
+	)
+
+	chrootDirPath := filepath.Join(s.BuildDirPath, chrootName)
+	s.chroot = safechroot.NewChroot(chrootDirPath, existingDir)
+	s.chrootSpecDir = chrootSpecDirPath
+
+	extraDirectories := []string{}
+	extraMountPoints := []*safechroot.MountPoint{
+		safechroot.NewMountPoint(specsDirPath, s.chrootSpecDir, "", safechroot.BindMountPointFlags, ""),
+	}
+	err = s.chroot.Initialize(s.WorkerTarPath, extraDirectories, extraMountPoints)
+	if err != nil {
+		logger.Log.Errorf("Failed to initialize chroot (%s) inside (%s). Error: %v.", s.WorkerTarPath, chrootDirPath, err)
+	}
+
+	return
+}
+
+// CleanUp cleans up the chroot environment. This function should be called after all other functions in this package have been
+// called (likely in a defer statement)
+func (s *SimpleChrootTool) CleanUp() {
+	const leaveFilesOnDisk = false
+
+	if s.chroot != nil {
+		s.chroot.Close(leaveFilesOnDisk)
+	}
+}
+
+// BuildDefines creates a map of RPM build defines for the crhoot. This function should be called after InitializeChroot()
+// and from a context inside a call to the chroot's .Run() function.
+func (s *SimpleChrootTool) BuildDefines(distTag string) map[string]string {
+	const runCheck = true
+
+	defines := rpm.DefaultDefines(runCheck)
+	defines[rpm.DistTagDefine] = distTag
+
+	return defines
+}
+
+// BuildCompatibleSpecsList builds a list of spec files in the chroot's SPECs directory that are compatible with the build arch. Paths
+// are relative to the chroot's base directory. This function should be called after InitializeChroot() and from a context inside a call
+// to the chroot's .Run() function.
+func (s *SimpleChrootTool) BuildCompatibleSpecsList(inputSpecPaths []string, defines map[string]string) (filteredSpecPaths []string, err error) {
```

Comment:
>, defines map[string]strin [](http://example.com/codeflow?start=75&length=26)

For public functions, let's remove the `defines` argument and add an optional `SetDefines` method.

Reply:
With moving the spec functions to a new package this no longer makes sense. Will need to pass along.